force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -57,8 +65,19 @@ public with sharing class FlowControlLoopsCollections {
     public static Integer findMax(Integer num1, Integer num2) {
         // Initialize largestNum as null
         Integer largestNum = null;
+        if(num1 != null && num2 != null){
+            if(num1 == num2) {
+                largestNum = num1;
+            }
+            else if(num1 > num2) {
+                largestNum = num1;
+            }
+            else {
+                largestNum = num2;
+            }    
+        }
```

Comment:
Here again, instead of nesting your if statements within an if statement, we want to identify invalid input, and return early. 

if num1 == null || num2 == null